go-client/client/tablehandle.go

```diff
@@ -0,0 +1,326 @@
+// Columns not used in the grouping must be numeric.
+func (th *TableHandle) StdBy(ctx context.Context, cols ...string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, cols, "", tablepb2.ComboAggregateRequest_STD)
+}
+
+// VarBy returns the variance for each group. Null values are ignored.
+// Columns not used in the grouping must be numeric.
+func (th *TableHandle) VarBy(ctx context.Context, cols ...string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, cols, "", tablepb2.ComboAggregateRequest_VAR)
+}
+
+// MedianBy returns the median value for each group. Null values are ignored.
+// Columns not used in the grouping must be numeric.
+func (th *TableHandle) MedianBy(ctx context.Context, cols ...string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, cols, "", tablepb2.ComboAggregateRequest_MEDIAN)
+}
+
+// MinBy returns the minimum value for each group. Null values are ignored.
+// Columns not used in the grouping must be numeric.
+func (th *TableHandle) MinBy(ctx context.Context, cols ...string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, cols, "", tablepb2.ComboAggregateRequest_MIN)
+}
+
+// MaxBy returns the maximum value for each group. Null values are ignored.
+// Columns not used in the grouping must be numeric.
+func (th *TableHandle) MaxBy(ctx context.Context, cols ...string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, cols, "", tablepb2.ComboAggregateRequest_MAX)
+}
+
+// CountBy returns the number of rows for each group.
+// The count of each group is stored in a new column named after the resultCol argument.
+func (th *TableHandle) CountBy(ctx context.Context, resultCol string, cols ...string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, cols, resultCol, tablepb2.ComboAggregateRequest_COUNT)
+}
+
+// Count counts the number of values in the specified column and returns it as a table with one row and one column.
+func (th *TableHandle) Count(ctx context.Context, col string) (*TableHandle, error) {
+	return th.client.dedicatedAggOp(ctx, th, nil, col, tablepb2.ComboAggregateRequest_COUNT)
+}
+
+// AggBy applies a list of aggregations to table data.
+// See the docs on AggBuilder for details on what each of the aggregation types do.
+func (th *TableHandle) AggBy(ctx context.Context, agg *AggBuilder, by ...string) (*TableHandle, error) {
+	return th.client.aggBy(ctx, th, agg, by)
+}
+
+// Merge combines two or more tables into one aggregate table.
+// This essentially appends the tables one on top of the other.
+// If sortBy is provided, the resulting table will be sorted based on that column.
+func (th *TableHandle) Merge(ctx context.Context, sortBy string, others ...*TableHandle) (*TableHandle, error) {
+	tables := make([]*TableHandle, len(others)+1)
+	tables[0] = th
+	copy(tables[1:], others)
+
+	return th.client.merge(ctx, sortBy, tables)
+}
```

Comment:
Should it be a method on `Client`, or a freestanding function? In either case, this means we'll need to rename the query version of this method to something like `MergeQuery` to avoid conflicts.

Reply:
Related note: If `Merge` takes a list of table handles instead of a handle + a list, then now we have to return an error if the user passes in zero tables. Thus, `Merge` can't be chained anymore (which is mostly an issue for the query Merge). Is this acceptable?

Reply:
I was able to fix the query merge so that it can actually return the error when the query is built. This is no longer a problem.